7HW/2_task.js

```diff
@@ -0,0 +1,56 @@
+"use strict";
+let num = 1;
+logSuperCool();
+const characters = [
+    { 'name': 'Barney', 'age': 36 },
+    { 'name': 'Fred', 'age': 40 },
+    { 'name': 'Jack', 'age': 50 }
+  ];
+// 1. Напишите функцию addCharacter(character) позволяющую добавить новый объект в массив characters. 
+//     Объект должен иметь поля name (string) и age (number)
+function addCharacter(character = {}) {
+    if ('name' in character && 'age' in character) {
+        if (typeof character['name'] === 'string' && typeof character['age'] === 'number') {
+        characters.push(character);
+        } else console.log('Поля не того формата');
+    } else console.log('Отсутствуют нужные поля');
+}
```

Comment:
```suggestion
function addCharacter(character = {}) {
    if ('name' in character && 'age' in character && typeof character['name'] === 'string' && typeof character['age'] === 'number') {
     characters.push(character);
       } else {
throw new Error("Invalid character")
}
}
```
Пишу с телефона, потому текстом добавлю: в элс- синтаксис пробрасывания ошибки. И лучше это делать сразу в верху функции в иф, а позитивный сценарий в элс